urequests/urequests.py

```diff
@@ -33,6 +33,9 @@ def json(self):
 
 
 def request(method, url, data=None, json=None, headers={}, stream=None):
+    if auth is not None:
+        headers.update(encode_basic_auth(auth))
```

Comment:
* An `auth` parameter needs to be added to the `request` function signature (defaulting to `None`).

* If following my suggested changes for `encode_basic_auth`, this should be changed to the following then:

      if auth:
          headers.update(encode_basic_auth(auth[0], auth[1]))

  (Also changing `if auth is not None` to `if auth` so that passing an empty tuple works as well.)

Reply:
It might be also a good idea to allow passing a callable for auth (but this would be an extension not supported by the original `requests.request()`.

    if auth:
        headers.update(auth() if callable(auth) else encode_basic_auth(auth[0], auth[1]))

Reply:
Obviously that mistake had to happen to me 🤦
Seems like I messed up when developing...

Reply:
> ... this would be an extension not supported by the original `requests.request()`.

How are we dealing with that? Do we actually want to extend the original or simply replicate it's functionality? Is there any guidelines for that?